items/managed.js

```diff
@@ -248,6 +249,51 @@ class Item {
         }
         managedItemProvider.update(this.rawItem);
     }
+
+    /**
+     * Converts the Item's state to a time.ZonedDateTime if possible.
+     * - PercentType: treated as a number of seconds added to now
+     * - DecimalType: rounded to the nearest integer and added as milliseconds to now
+     * - QuantityType<Time>: rounded to the nearest second and added to now
+     * - StringType:
+     *   - HH:mm:ss: 24hr time, date time at the given time with today's date
+     *   - hh:mm:ss aa: 12hr time, date time at the given time with today's date
+     *   - duration string: see the docs for js-joda Duration:
+     * - DateTimeType: converted to a time.ZonedDateTime
+     * @returns {time.ZonedDateTime} the state converted to a date time
+     * @throws will throw an error if the state cannot be converted to a date time
+     */
+    toZonedDateTime() {
+        if(this.rawItem.state instanceof PercentType) {
```

Comment:
Out of curiosity, when would you use a percentType as seconds?  Not a use case i'm familiar with